devito/ops/node_factory.py

```diff
@@ -16,6 +16,8 @@ class OPSNodeFactory(object):
 
     def __init__(self):
         self.ops_args = OrderedDict()
+        self.ops_args_accesses = defaultdict(list)
+        self.ops_params = []
```

Comment:
is it ok that a factory class keeps state?

Reply:
This greatly simplifies the next steps

Reply:
oki, as long as factories are never used as instance attributes, I'm fine with it